src/FragmentCache.php

```diff
@@ -4,48 +4,48 @@
 
 namespace Yiisoft\Yii\Widgets;
 
-use function array_merge;
-use function ob_get_clean;
-use function ob_implicit_flush;
-use function ob_start;
+use RuntimeException;
 use Yiisoft\Cache\CacheInterface;
 use Yiisoft\Cache\Dependency\Dependency;
-
-use Yiisoft\View\DynamicContentAwareInterface;
-use Yiisoft\View\DynamicContentAwareTrait;
-use Yiisoft\View\WebView;
+use Yiisoft\View\Cache\CacheContent;
+use Yiisoft\View\Cache\DynamicContent;
 use Yiisoft\Widget\Widget;
 
-final class FragmentCache extends Widget implements DynamicContentAwareInterface
-{
-    use DynamicContentAwareTrait;
+use function ob_end_clean;
+use function ob_get_clean;
+use function ob_implicit_flush;
+use function ob_start;
 
-    private string $id;
+final class FragmentCache extends Widget
+{
+    private ?string $id = null;
     private CacheInterface $cache;
-    private int $duration = 60;
     private ?Dependency $dependency = null;
-    private $variations;
-    private ?string $content = null;
+    private int $ttl = 60;
+    private array $variations = [];
 
-    private WebView $webView;
+    /**
+     * @var array<string, DynamicContent>
```

Comment:
```suggestion
     * @var DynamicContent[]
```

Reply:
No. Then you can add different objects with the same `ID`.